Bump dompurify, handsontable, and @testing-library/react

alesan99 opened the pull request:
Fixes #7322

Some types changed so some minor code changes were needed.
Upgraded @testing-library/react too because multiple tests were suddenly failing (because user events weren't being automatically wrapped in act() ).

### Checklist

- [x] Self-review the PR after opening it to make sure the changes look good and
      self-explanatory (or properly documented)
- [ ] Add relevant issue to release milestone
- [ ] Add pr to documentation list
- [x] Add automated tests

### Testing instructions

- Go to a workbench dataset
- [ ] Make sure no new errors occur when adding and modifying data on the spreadsheet.
- [ ] Make sure scrolling/navigation works as expected when you have a lot of rows or columns. (Like this dataset:  [FishTissueData.tsv](https://github.com/user-attachments/files/22471019/FishTissueData.tsv) )
- [ ] Make sure workbench performance/speed when scrolling isn't worse than in main.
- [ ] Make sure you can successfully validate and upload a dataset.

alesan99 committed:
Update handsontable and wrapper

alesan99 committed:
Update handsontable and wrapper

alesan99 committed:
Update package-lock

alesan99 committed:
Update types

alesan99 committed:
Update types in WbSpreadsheet

alesan99 committed:
Update fast-xml-parser types
Fix missing HotTable type

alesan99 committed:
Merge branch 'main' into issue-7309

alesan99 committed:
WIP update test

alesan99 committed:
undo test changes

alesan99 committed:
Upgrade @testing-library/react to automatically wrap user-events with act()

alesan99 committed:
Explicitly set height of HOT's container

bhumikaguptaa reviewed (approved):
- [X] Make sure no new errors occur when adding and modifying data on the spreadsheet.
- [X] Make sure scrolling/navigation works as expected when you have a lot of rows or columns.
- [X] Make sure you can successfully validate and upload a dataset.

Looks good!

emenslin reviewed (changes requested):
- [x] Make sure no new errors occur when adding and modifying data on the spreadsheet.
- [ ] Make sure scrolling/navigation works as expected when you have a lot of rows or columns.
- [x] Make sure you can successfully validate and upload a dataset.

---
Scrolling behavior is a bit weird and rows/columns seem to constantly change size. 

Main: 

https://github.com/user-attachments/assets/7ee8b987-2492-4b90-9070-1bafa6670432


Issue branch: 

https://github.com/user-attachments/assets/a3c5a714-e6bd-4a86-aa9c-7b035c0a4da2

alesan99 committed:
Merge branch 'main' into issue-7309

alesan99 committed:
Lint code with ESLint and Prettier

Triggered by fca951ab67599538d704edf5e4e33a4bb170ec4d on branch refs/heads/issue-7309

alesan99 committed:
Force column and rendering updates

alesan99 committed:
Lint code with ESLint and Prettier

Triggered by e63a043b1b6557c83cae0fb4d078471db63588da on branch refs/heads/issue-7309

alesan99 committed:
Upgrade handsontable to 16.1.0

alesan99 committed:
Merge branch 'issue-7309' of https://github.com/specify/specify7 into issue-7309

emenslin reviewed (changes requested):
- [x] Make sure no new errors occur when adding and modifying data on the spreadsheet.
- [ ] Make sure scrolling/navigation works as expected when you have a lot of rows or columns. (Like this dataset:  [FishTissueData.tsv](https://github.com/user-attachments/files/22471019/FishTissueData.tsv) )
- [ ] Make sure workbench performance/speed when scrolling isn't worse than in main.
- [x] Make sure you can successfully validate and upload a dataset.

---

Scrolling behavior is now lagging and sometimes does not render the rows. 

Issue branch: 

https://github.com/user-attachments/assets/92f1c734-5371-4cbe-b672-ca0fd176ddaf


Main:

https://github.com/user-attachments/assets/6fefc159-0a2e-471c-aa9b-4a47646fbb6d

alesan99 commented:
Hmm okay after some consideration it might be best to wait for handsontable to possibly address some of the rendering issues.

It looks like autoColumnSize is broken on virtualized tables, and trying to manually fix it makes it laggy so I'm not sure there's a good solution right now.

This PR can be repurposed once a new update is out.
